Modules/Filtering/BiasCorrection/include/itkN4BiasFieldCorrectionImageFilter.hxx

```diff
@@ -85,25 +124,28 @@ N4BiasFieldCorrectionImageFilter<TInputImage, TMaskImage, TOutputImage>
 
   ImageAlgorithm::Copy( inputImage, logInputImage.GetPointer(), inputRegion, inputRegion);
 
-  const MaskImageType * maskImage = this->GetMaskImage();
-  const RealImageType * confidenceImage = this->GetConfidenceImage();
+  const auto maskImageRange = MakeImageRange(this->GetMaskImage());
+  const auto confidenceImageRange = MakeImageRange(this->GetConfidenceImage());
   const MaskPixelType maskLabel = this->GetMaskLabel();
   const bool useMaskLabel = this->GetUseMaskLabel();
 
-  ImageRegionIteratorWithIndex<RealImageType> It( logInputImage, inputRegion );
+  const ImageRange<RealImageType> logInputImageRange{ *logInputImage };
```

Comment:
It is important that the regions the iterator is operating on is explicit. It will make the operation clear, and perhaps help with some of the other regions related problems that may be manifesting themselves.